src/weather.cpp

```diff
@@ -466,6 +467,9 @@ void weather_effect::thunder()
         } else if( g->u.has_trait( trait_GOODHEARING ) && one_in( 1 - 2 * g->get_levz() ) ) {
             add_msg( _( "You hear a rumble of thunder from above." ) );
             sfx::play_variant_sound( "environment", "thunder_far", 100, rng( 0, 359 ) );
+        } else if( g->u.has_trait( trait_GOODHEARING2 ) && one_in( 1 - 1 * g->get_levz() ) ) {
+            add_msg( _( "You hear a rumble of thunder from above." ) );
+            sfx::play_variant_sound( "environment", "thunder_far", 120, rng( 0, 359 ) );
```

Comment:
This should use hearing modifier from mutations instead of checking for certain traits.

Reply:
I tried to keep it the same way that good hearing and bad hearing works. Because bad hearing doesnt work that way it is not in the scope of this pull request.

Reply:
These checks should include animal ears and all other hearing mutation, otherwise it can be considered as a bug.
It can be simplified by using `Character::mutation_value( "hearing_modifier" )` which will return hearing modifier from all mutations.

Reply:
Very true, I have removed the code from this PR. I separate PR would be needed to change it to how it should be.

Reply:
Dont see any bad if a feature PR will fix some related bugs.
Actually I should do that recently when I unhardcoded hearing mutations, but I did not know about these checks.
This code can be replaced with something like
```
} else if( one_in( 2 + std::max( - roll_remainder( Character::mutation_value( "hearing_modifier" ) ) - g->get_levz() , -1 ) ) {
    add_msg( _( "You hear a rumble of thunder from above." ) );
    sfx::play_variant_sound( "environment", "thunder_far", 100, rng( 0, 359 ) );
}
```

Reply:
Would this Work?
```
} else if( one_in( 2 + std::max( - roll_remainder( Character::mutation_value ( "hearing_modifier" ) ) - g->get_levz() , -1 ) ) ) {
            add_msg( _( "You hear a rumble of thunder from above." ) );
            sfx::play_variant_sound( "environment", "thunder_far", (80 * Character::mutation_value( "hearing_modifier" ) ) , rng( 0, 359 ) );
        } 
```

Reply:
Z level depth should increase value in `one_in` and make rumbles more difficult to hear, while hearing mutations should decrease it, but not make it zero or negative, that is why I used `std::max`.
Anyway this change has very minor impact on the gameplay, so any other realization will be fine too.

Reply:
This Works, letting a very good hearing modifier always hear the rumble at lower z levels.
```
}  else if( one_in( std::max( round( 2 * g->get_levz() / Character::mutation_value( "hearing_modifier" ) ) ) , 1 ) ) {
            add_msg( _( "You hear a rumble of thunder from above." ) );
            sfx::play_variant_sound( "environment", "thunder_far", ( 80 * Character::mutation_value( "hearing_modifier" ) ) , rng( 0, 359 ) );
} 
```

Reply:
This Issue is now in its own PR at #28857